test/e2e/reconciler/cache/replication_test.go

```diff
@@ -144,225 +163,195 @@ func replicateAPIResourceSchemaNegativeScenario(ctx context.Context, t *testing.
 	t.Helper()
-
-	org := framework.NewOrganizationFixture(t, server)
-	cluster := framework.NewWorkspaceFixture(t, server, org.Path(), framework.WithRootShard())
-	resourceName := "wild.wild.west"
-	scenario := &replicateResourceScenario{resourceName: resourceName, resourceKind: "APIExport", server: server, kcpShardClusterClient: kcpShardClusterClient, cacheKcpClusterClient: cacheKcpClusterClient}
-
-	t.Logf("Create source APIExport %s/%s on the root shard for replication", cluster, resourceName)
-	scenario.CreateSourceResource(t, func() error {
-		apifixtures.CreateSheriffsSchemaAndExport(ctx, t, cluster.Path(), kcpShardClusterClient, "wild.wild.west", "testing replication to the cache server")
-		return nil
-	})
-	t.Logf("Verify that the source APIExport %s/%s was replicated to the cache server", cluster, resourceName)
-	scenario.VerifyReplication(ctx, t, cluster.Path())
-
-	t.Logf("Change the spec on source APIExport %s/%s and verify if updates were propagated to the cached object", cluster, resourceName)
-	scenario.UpdateSourceResource(ctx, t, cluster.Path(), func(res runtime.Object) error {
-		apiExport, ok := res.(*apisv1alpha1.APIExport)
-		if !ok {
-			return fmt.Errorf("%T is not *APIExport", res)
-		}
-		apiExport.Spec.LatestResourceSchemas = append(apiExport.Spec.LatestResourceSchemas, "foo.bar")
-		_, err := kcpShardClusterClient.Cluster(cluster.Path()).ApisV1alpha1().APIExports().Update(ctx, apiExport, metav1.UpdateOptions{})
-		return err
-	})
-	scenario.VerifyReplication(ctx, t, cluster.Path())
-
-	t.Logf("Change some metadata on source APIExport %s/%s and verify if updates were propagated to the cached object", cluster, resourceName)
-	scenario.UpdateSourceResource(ctx, t, cluster.Path(), func(res runtime.Object) error {
-		if err := scenario.ChangeMetadataFor(res); err != nil {
-			return err
-		}
-		apiExport, ok := res.(*apisv1alpha1.APIExport)
-		if !ok {
-			return fmt.Errorf("%T is not *APIExport", res)
-		}
-		_, err := kcpShardClusterClient.Cluster(cluster.Path()).ApisV1alpha1().APIExports().Update(ctx, apiExport, metav1.UpdateOptions{})
-		return err
-	})
-	scenario.VerifyReplication(ctx, t, cluster.Path())
-
-	t.Logf("Verify that deleting source APIExport %s/%s leads to removal of the cached object", cluster, resourceName)
-	scenario.DeleteSourceResourceAndVerify(ctx, t, cluster.Path())
+	replicateResource(ctx,
+		t,
+		server,
+		kcpShardClusterDynamicClient,
+		cacheKcpClusterDynamicClient,
+		"",
+		"wild.wild.west",
+		"APIExport",
+		apisv1alpha1.SchemeGroupVersion.WithResource("apiexports"),
+		&apisv1alpha1.APIExport{ObjectMeta: metav1.ObjectMeta{Name: "wild.wild.west"}},
+		&apisv1alpha1.APIExport{ObjectMeta: metav1.ObjectMeta{Name: "wild.wild.west"}, Spec: apisv1alpha1.APIExportSpec{LatestResourceSchemas: []string{"foo.bar"}}},
```

Comment:
I feel like having the resource definitions outside of the call to replicateResource may improve the readability.

Reply:
yes, maybe we could bring them closer to the place where we define scenarios

Reply:
i'm not sure exactly how it should look, I will try to think about it and change it on the next iteration

Reply:
For now could you have the fields on new line like in replicateAPIExportNegativeScenario?

Reply:
done, ptal